crates/utils/src/request.rs

```diff
@@ -47,27 +47,31 @@ where
   response.expect("retry http request")
 }
 
-#[derive(Deserialize, Debug)]
-pub(crate) struct IframelyResponse {
-  title: Option<String>,
-  description: Option<String>,
+#[derive(Deserialize, Debug, Default)]
+pub struct IframelyResponse {
+  pub title: Option<String>,
+  pub description: Option<String>,
   thumbnail_url: Option<Url>,
-  html: Option<String>,
+  pub html: Option<String>,
 }
 
 pub(crate) async fn fetch_iframely(
   client: &Client,
   url: &Url,
 ) -> Result<IframelyResponse, LemmyError> {
-  let fetch_url = format!("{}/oembed?url={}", Settings::get().iframely_url(), url);
+  if let Some(iframely_url) = Settings::get().iframely_url {
+    let fetch_url = format!("{}/oembed?url={}", iframely_url, url);
 
-  let response = retry(|| client.get(&fetch_url).send()).await?;
+    let response = retry(|| client.get(&fetch_url).send()).await?;
 
-  let res: IframelyResponse = response
-    .json()
-    .await
-    .map_err(|e| RecvError(e.to_string()))?;
-  Ok(res)
+    let res: IframelyResponse = response
+      .json()
+      .await
+      .map_err(|e| RecvError(e.to_string()))?;
+    Ok(res)
+  } else {
+    Ok(IframelyResponse::default())
```

Comment:
Instead of returning a default here, this should be an error, maybe an anyhow saying that iframely hasn't been set up.